api-lib/event_triggers/utils/handleVouchMsg.ts

```diff
@@ -1,8 +1,57 @@
 import * as queries from '../../../api-lib/gql/queries';
+import { isFeatureEnabled } from '../../../src/config/features';
 import { NotFoundError } from '../../HttpError';
-import { sendSocialMessage } from '../../sendSocialMessage';
+import {
+  Channels,
+  DiscordVouch,
+  sendSocialMessage,
+} from '../../sendSocialMessage';
+import { Awaited } from '../../ts4.5shim';
 import { EventTriggerPayload } from '../../types';
 
+type GetChannelsProps = {
+  channels: Channels<DiscordVouch>;
+  circle: Awaited<ReturnType<typeof queries.getCircle>>['circles_by_pk'];
+  nominee: Awaited<ReturnType<typeof queries.getNominee>>['nominees_by_pk'];
+  vouch: Awaited<ReturnType<typeof queries.getExistingVouch>>['vouches'][0];
+};
+
+function getChannels(props: GetChannelsProps): Channels<DiscordVouch> {
+  const {
+    channels,
+    vouch: { voucher },
+    nominee,
+    circle,
+  } = props || {};
+
+  if (isFeatureEnabled('discord') && channels?.discordBot) {
+    const { discord_channel_id: channelId, discord_role_id: roleId } =
+      circle?.discord_circle || {};
+
+    if (!channelId || !roleId) {
+      return null;
+    }
+
+    return {
+      discordBot: {
+        type: 'vouch' as const,
+        channelId,
+        roleId,
+        nominee: nominee?.profile?.name,
+        voucher: voucher?.profile.name ?? voucher?.name ?? 'Someone',
+        nominationReason: nominee?.description ?? 'unknown reason',
+        currentVouches: Math.max(
+          0,
+          (nominee?.nominations_aggregate.aggregate?.count ?? 0) - 1
+        ),
```

Comment:
@topocount is this correct? Assuming that a vouch is nothing but a nomination, so the nomination itself would be a vouch, but the required vouches do not included the said initial nomination.

Also, a bit uncertain what to do in case we don't have some data, for now just using placeholder values.

Reply:
This looks fine. What are the conditions where we'd be missing some data that you're thinking of.

Reply:
I was just uncertain about these:
```typescript
voucher: voucher?.profile.name ?? voucher?.name ?? 'Someone',
nominationReason: nominee?.description ?? 'unknown reason',
```
If for some reason we are missing the voucher name or the nomination reason, but AFAIK these are and always were required?

Reply:
I think this is just an issue of joins. The edge cases where this data wouldn't be available are remote.